extensions/ui-extension/src/fields/Newsletter.tsx

```diff
@@ -0,0 +1,91 @@
+import {
+   Form,
+   Button,
+   BlockStack,
+   Heading,
+   TextField,
+   Text,
+   useTranslate,
+   Grid,
+   Banner,
+} from '@shopify/ui-extensions-react/checkout';
+import { useState } from 'react';
+
+const ifixit_base_url = 'https://www.ifixit.com';
+
+const emailRegex = /^([a-zA-Z0-9._%-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,})$/;
+
+function isValidEmail(email: string) {
+   return emailRegex.test(email);
+}
+
+export function Newsletter() {
+   const [showEntryBox, setShowEntryBox] = useState(true);
+   const [showError, setShowError] = useState(false);
+   const [validationError, setValidationError] = useState('');
+   const [isLoading, setShowLoading] = useState(false);
+   let email = '';
```

Comment:
The email should be state unless there's a good reason not to

```suggestion
   const [email, setEmail] = useState('');
```